Lib/pickle.py

```diff
@@ -188,8 +187,10 @@ def __init__(self, value):
 NEXT_BUFFER      = b'\x97'  # push next out-of-band buffer
 READONLY_BUFFER  = b'\x98'  # make top of stack readonly
 
-__all__.extend([x for x in dir() if re.match("[A-Z][A-Z0-9_]+$", x)])
-
+__all__.extend([
+    x for x in dir()
+    if x.isupper() and x.isidentifier() and not x.startswith('_')
+])
```

Comment:
Everything in `dir()` will be an identifier. To be more faithful to the regex you could add `x.isascii()`, but probably not needed. Hence:
```suggestion
__all__.extend(x for x in dir() if x.isupper() and not x.startswith('_'))
```